### PR TITLE
Hook up user search to the organisers in batch edit

### PR DESCRIPTION
**Pull request by bp323**
screen shot 2015-02-10 at 10 19 21

Create a batch edit component similar to the subheader pickers that allows searching for users in the system and adding them as an organiser. The same component should be used in the individual fields.

### CODE REVIEW

**Comment by bp323** [comment]
Reassigning to @Coenego for review

File: shared/gh/js/views/gh.admin-batch-edit.js [hunk 1]
```diff
@@ -605,42 +695,49 @@ define(['gh.api.event', 'gh.api.groups', 'gh.api.series', 'gh.api.util', 'gh.adm
...
                     'notes': $('.gh-event-type', $eventContainer).attr('data-type'),
-                    'organisers': $('.gh-event-organisers', $eventContainer).text() || null,
+                    'organiserOther': organisers.organiserOthers || null,
+                    'organiserUsers': organisers.organiserUsers || null,
                     'start': $('.gh-event-date', $eventContainer).attr('data-start')
                 };
                 newEventObjs.push(newEventObj);
             });
-
+console.log(newEventObjs);
```
Comment (Coenego) [inline shared/gh/js/views/gh.admin-batch-edit.js]: This can probably go

File: shared/gh/js/views/gh.admin-batch-edit.js [hunk 2]
```diff
@@ -595,6 +631,60 @@ define(['gh.api.event', 'gh.api.groups', 'gh.api.series', 'gh.api.util', 'gh.adm
...
+                organisers[userId] = toAdd;
+            } else {
+                organisers[displayName] = toAdd;
+            }
+        });
+
+        return organisers;
+    };
+
+    var getOrganiserList = function($container) {
```
Comment (Coenego) [inline shared/gh/js/views/gh.admin-batch-edit.js]: Function documentation needed

File: apps/timetable/admin/ui/css/admin.css [hunk 3]
```diff
@@ -864,3 +864,73 @@
...
+ul.as-selections li.as-selection-item,
+ul.as-selections li.as-selection-item a.as-close,
+ul.as-list {
+    font-family: 'Myriad Pro', helvetica, sans-serif;
+}
+
+.as-selections {
+    border-style: solid;
+    border-width: 1px;
+    border-top-width: 1px !important;
```
Comment (Coenego) [inline apps/timetable/admin/ui/css/admin.css]: `border-top-width` should go before `border-width`

File: apps/timetable/admin/ui/css/admin.css [hunk 4]
```diff
@@ -864,3 +864,73 @@
...
+}
+
+.as-selections .as-original {
+    margin: 0 !important;
+}
+
+.as-selections .as-original input {
+    font-size: 15px !important;
+    height: 100% !important;
+    border: none !important;
```
Comment (Coenego) [inline apps/timetable/admin/ui/css/admin.css]: `border` should go before `font-size`

File: apps/timetable/admin/ui/css/skin.css [hunk 5]
```diff
@@ -402,3 +402,55 @@
...
+
+.as-results .as-list .as-result-item.active {
+    background-color: #92308A !important;
+    color: #FFF !important;
+}
+
+.as-results .as-list .as-result-item.active em,
+.as-results .as-list .as-result-item em {
+    background-color: #412F5B !important;
+}
```
Comment (Coenego) [inline apps/timetable/admin/ui/css/skin.css]: Empty line missing at the bottom

File: shared/gh/js/gh.bootstrap.js [hunk 6]
```diff
@@ -68,8 +69,10 @@ requirejs.config({
         // GH view controllers
         'gh.admin-batch-edit': 'gh/js/views/gh.admin-batch-edit',
         'gh.admin-batch-edit-date': 'gh/js/views/gh.admin-batch-edit-date',
+        'gh.admin-batch-edit-organiser': 'gh/js/views/gh.admin-batch-edit-organiser',
         'gh.admin-event-type-select': 'gh/js/views/gh.admin-event-type-select',
         'gh.admin-listview': 'gh/js/views/gh.admin-listview',
+        'gh.admin-edit-organiser': 'gh/js/views/gh.admin-edit-organiser',
```
Comment (Coenego) [inline shared/gh/js/gh.bootstrap.js]: This line should go after `gh.admin-listview`

File: shared/gh/js/views/gh.admin-batch-edit-organiser.js [hunk 7]
```diff
@@ -0,0 +1,185 @@
+/*!
+ * Copyright 2014 Digital Services, University of Cambridge Licensed
```
Comment (Coenego) [inline shared/gh/js/views/gh.admin-batch-edit-organiser.js]: 2015 in the meantime

File: shared/gh/js/views/gh.admin-edit-organiser.js [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+         * Create an AutoSuggest component and attach it to the generated form.
+         * The form is available inside the function as variable this.
+         *
+         * @param  {Object}    settings    The jEditable field settings
+         * @param  {String}    original    The containing HTML element
+         * @private
+         */
+        'element' : function(settings, original) {
+            // Render the organiser template
+            var content = gh.api.utilAPI.renderTemplate($('#gh-organiser-template'), {
```
Comment (Coenego) [inline shared/gh/js/views/gh.admin-edit-organiser.js]: `gh.utils.renderTemplate`

File: shared/gh/js/views/gh.admin-batch-edit.js [hunk 9]
```diff
@@ -402,6 +427,22 @@ define(['gh.constants', 'gh.utils', 'gh.api.event', 'gh.api.groups', 'gh.api.ser
                 }
             }
         });
+
+        // Apply jEditable to the organisers
+        $('.gh-event-organisers').editable(editableOrganiserSubmitted, {
+            'cssclass' : 'gh-jeditable-form',
```
Comment (Coenego) [inline shared/gh/js/views/gh.admin-batch-edit.js]: Space between `'cssclass'` and `:` can probably go

File: shared/gh/js/views/gh.admin-batch-edit.js [hunk 10]
```diff
@@ -605,42 +695,49 @@ define(['gh.constants', 'gh.utils', 'gh.api.event', 'gh.api.groups', 'gh.api.ser
...
-                    // 'group': '',
                     'notes': $('.gh-event-type', $eventContainer).attr('data-type'),
-                    'organisers': $('.gh-event-organisers', $eventContainer).text() || null,
+                    'organisers': organisers || null,
                     'start': $('.gh-event-date', $eventContainer).attr('data-start')
                 };
                 updatedEventObjs.push(updatedEventObj);
             });
 
+
```
Comment (Coenego) [inline shared/gh/js/views/gh.admin-batch-edit.js]: This line can probably go

File: shared/gh/js/views/gh.admin-edit-organiser.js [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+ * permissions and limitations under the License.
+ */
+
+define(['gh.core', 'jquery.jeditable'], function(gh) {
+
+    /**
+     * Get all organisers that were previously added to the AutoSuggest field
+     *
+     * @param  {String}    original    The containing HTML element
+     * @private
```
Comment (Coenego) [inline shared/gh/js/views/gh.admin-edit-organiser.js]: `@return` missing

File: shared/gh/js/views/gh.admin-edit-organiser.js [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+                    'id': 'gh-event-select-' + String(Math.ceil(Math.random() * 10000))
+                }
+            });
+            $(this).append(content);
+            return this;
+        },
+
+        /**
+         * Set the default jEditable field content
+         * @private
```
Comment (Coenego) [inline shared/gh/js/views/gh.admin-edit-organiser.js]: Extra line needed in between

**Comment by Coenego** [comment]
Reassigning to @bp323 for follow-up

**Comment by bp323** [comment]
Reassigning to @Coenego for review

**Comment by Coenego** [comment]
Code looks good!

General thoughts: 
- When adding multiple organisers to an event, the seem to be appearing under each other, which can be confusing since you can't see the previously added organisers. 
- It should also be possible to add matching results by pressing the tab button. Currently only the current entered string is added to the organisers.

Reassigning to @bp323

**Pull request merged**